### ISSUE TITLE
Work Package Templates - Delete Endpoint

### ISSUE THREAD

**Issue by RChandler234** [Northeastern-Electric-Racing/FinishLine#2154]
### Description

Blocked By #2148 

We need an endpoint so that users can delete work package templates in the database. 

### Acceptance Criteria

- [ ] Add all these things in the work package files
- [ ] This endpoint should be on the route '/template/:workpackageTemplateId/delete'
- [ ] You should be 'Soft Deleting' the work package template by setting its dateDeleted and userDeleted
- [ ] Only admins should be able to delete work package templates
- [ ] If this template is referenced as a blocked by in another template, you should throw an error
- [ ] Backend tests will be added in another ticket

### Proposed Solution

Look at other delete endpoints for an example
